src/raised-button.jsx

```diff
@@ -364,11 +368,10 @@ const RaisedButton = React.createClass({
 
     const iconCloned = icon && React.cloneElement(icon, {
       color: styles.label.color,
-      style: {
-        verticalAlign: 'middle',
-        marginLeft: labelPosition === 'before' ? 0 : 12,
-        marginRight: labelPosition === 'before' ? 12 : 0,
-      },
+      style: Object.assign({}, styles.icon, {
```

Comment:
the extra `{}` isn't really needed as `styles.icon` is not used anywhere else so: `Object.assign(styles.icon, { ...`.